### PR TITLE
Update kdocs for drop-in

### PR DESCRIPTION
**Pull request by jreij**
## Description
Update kdocs for drop-in

## Checklist 
- [x] Changes are tested manually

COAND-724

### CODE REVIEW

**Comment by jreij** [comment]
This PR does not cover all the classes in drop-in, it's still missing a few. Feel free to change anything you want and complete the PR.

File: drop-in/src/main/java/com/adyen/checkout/dropin/DropIn.kt [hunk 1]
```diff
@@ -62,91 +71,85 @@ object DropIn {
...
+     * Use [dropInConfiguration] to configure Drop-in and the components that will be loaded inside it.
      *
-     * You should always handle the cases of cancellation and error in [DropInCallback.onDropInResult].
+     * Optionally, you can extend [SessionDropInService] with your own implementation and add it to your manifest file.
+     * This allows you to interact with Drop-in, and take over the checkout flow.
      *
      * @param context The context to start the Checkout flow with.
      * @param dropInLauncher A launcher to start Drop-in, obtained with [registerForDropInResult].
-     * @param paymentMethodsApiResponse The result from the paymentMethods/ endpoint.
+     * @param checkoutSession The result from the /sessions endpoint. Use [CheckoutSessionProvider] to create this.
```
Comment (OscarSpruit) [inline drop-in/src/main/java/com/adyen/checkout/dropin/DropIn.kt:87]: Shouldn't `/sessions` be `/sessions/{id}/setup`?

File: drop-in/src/main/java/com/adyen/checkout/dropin/service/SessionDropInServiceContract.kt [hunk 2]
```diff
@@ -11,13 +11,184 @@ package com.adyen.checkout.dropin.service
...
+     *
+     * Once you take over the flow you will need to handle all the necessary subsequent network calls, otherwise a
+     * [MethodNotImplementedException] will be thrown. You can use the [isFlowTakenOver] field to check whether you did
+     * take over the flow in a previous call or not.
+     *
+     * We provide a [PaymentComponentState] which contains information about the state of the payment component at the
+     * moment the user submits the payment.
+     *
+     * We also provide inside [PaymentComponentState.data] the parameters that we can infer from the component's
+     * configuration and the user input, especially the *paymentMethod object with the shopper input details.
```
Comment (OscarSpruit) [inline drop-in/src/main/java/com/adyen/checkout/dropin/service/SessionDropInServiceContract.kt:37]: `*paymentMethod` is the star meant to make it bold or is it a mistake?

File: drop-in/src/main/java/com/adyen/checkout/dropin/DropIn.kt (hunk 1 above)
Comment (jreij) [reply drop-in/src/main/java/com/adyen/checkout/dropin/DropIn.kt:87]: `/sessions/{id}/setup` is a checkout shopper endpoint, they don't call it. They call `/sessions` and pass the response to `CheckoutSessionProvider` then we call `/sessions/{id}/setup`.

Does this make it any clearer?
`@param checkoutSession The result from the /sessions endpoint passed onto [CheckoutSessionProvider] to create this object.`

File: drop-in/src/main/java/com/adyen/checkout/dropin/service/SessionDropInServiceContract.kt (hunk 2 above)
Comment (jreij) [reply drop-in/src/main/java/com/adyen/checkout/dropin/service/SessionDropInServiceContract.kt:37]: It was written like `"paymentMethod"` before, i wanted to change it to `*paymentMethod*` but I forgot the other `*` 😅 . Maybe I should just write `the [state.data.paymentMethod] object`. WDYT?

Comment (OscarSpruit) [reply drop-in/src/main/java/com/adyen/checkout/dropin/service/SessionDropInServiceContract.kt:37]: I like this one `[state.data.paymentMethod]`

File: drop-in/src/main/java/com/adyen/checkout/dropin/DropIn.kt (hunk 1 above)
Comment (OscarSpruit) [reply drop-in/src/main/java/com/adyen/checkout/dropin/DropIn.kt:87]: Ohh I wasn't aware of that endpoint. Makes sense now

**Review by OscarSpruit — APPROVED** [review]

**Pull request merged**